docs/misc/commands.md

```diff
@@ -0,0 +1,211 @@
+ - `net.minecraft.commands.arguments` for all other argument types. Their names are always `<type>Argument`.
+ - `net.neoforged.neoforge.server.command`: for mod-oriented argument like `EnumArgument`. Their names are the same as for `net.minecraft`.
+
+All these argument types are implementing an interface: `ArgumentType<T>` where `T` is the type on which they are based.
+
+When you create an argument for your commands, you will always need to choose an `ArgumentType`.
+
+### Command Contexts
+Command contexts are used every time you run a command. It's based on a generic type, the additional information, often `CommandSourceStack`.
+It contains all the additional information plus what was written as arguments.
+
+### Exceptions
+The package `com.mojang.brigadier.exceptions` provide one [exception](https://docs.oracle.com/javase/tutorial/essential/exceptions/index.html): `CommandSyntaxException`, will be thrown if the syntax is incorrect or if the command fails. It will give an error message in chat if it was a player executing the command, or as output in a [command block](https://minecraft.wiki/w/command_block) if it's a command block.
+
+It also provides many internal handler classes and the `CommandExceptionType`s.
+
+#### `CommandExceptionType`
+This is an empty interface is used to mark a class as a eponymous exception. `CommandSyntaxException` take an argument `CommandExceptionType`.
+
+The classes implementing `CommandExceptionType` aims to create exception based on a certain amount of parameters:*
+ - `DynamicCommandExceptionType`: one
+ - `Dynamic2CommandExceptionType`, `Dynamic3CommandExceptionType`, `Dynamic4CommandExceptionType`: 2, 3 or 4
+ - `DynamicNCommandExceptionType`: an undetermined amount
+ - `SimpleCommandExceptionType`: zero
+
+These classes are most used in `BuiltInExceptions` in lines like:
+```java
+private static final Dynamic2CommandExceptionType FLOAT_TOO_BIG = new Dynamic2CommandExceptionType((found, max) -> new LiteralMessage("Float must not be
+    more than " + max + ", found " + found));
+```
+and then
+```java
+public Dynamic2CommandExceptionType floatTooHigh() {
+    return FLOAT_TOO_BIG;
+}
+```
+
+:::tip
+These functions are accessible everywhere, and it's recommended to use it when throwing `CommandSyntaxException`.
+```java
+throw new CommandSyntaxException(BuiltInException.floatTooHigh(), new LiteralMessage("Please provide a smaller float!"));
+```
+:::
+
+### `SharedSuggestionProvider`
+It's the base interface of execution contexts, [client-side or server-side](../concepts/sides.md).
+
+:::warning
+We talk here about **execution** context. It's meaning that the command hasn't be written when it's instanciated, so you can't get arguments from it, use instead [`CommandContext`](#command-contexts).
+:::
+
+#### `ClientSuggestionProvider`
+It implement `SharedSuggestionProvider` but have many more function, about client context.
```

Comment:
This does not explain what the client suggestion provider is. I don't even think it needs its own section, as it should be combined with either the client commands docs or how the command system works.